src/ucp/wireup/wireup.c

```diff
@@ -975,6 +967,82 @@ ucp_wireup_get_reachable_mds(ucp_worker_h worker,
     key->reachable_md_map = dst_md_map;
 }
 
+static void
+ucp_wireup_reclaim_unused_uct_ep(ucp_ep_h ep, ucp_ep_config_key_t *new_key)
+{
+    uct_ep_h new_uct_eps[UCP_MAX_LANES] = {NULL};
+    ucp_lane_index_t reuse_lane_map[UCP_MAX_LANES] = {UCP_NULL_LANE};
+    ucp_ep_config_key_t *old_key = &ucp_ep_config(ep)->key;
+    ucp_lane_index_t lane, reuse_lane;
+    ucs_queue_head_t uct_ep_req_queue;
+    ucp_request_t *req;
+
+    if ((ep->cfg_index == 0) || (old_key->num_lanes == new_key->num_lanes)) {
+        return;
+    }
+
+    for (lane = 0; lane < old_key->num_lanes; ++lane) {
+        ucs_assert(ep->uct_eps[lane] != NULL);
+        reuse_lane_map[lane] = UCP_NULL_LANE;
+        for (reuse_lane = 0; reuse_lane < new_key->num_lanes; ++reuse_lane) {
+            if ((old_key->lanes[lane].rsc_index ==
+                 new_key->lanes[reuse_lane].rsc_index)&&
+                (old_key->lanes[lane].path_index ==
+                 new_key->lanes[reuse_lane].path_index) &&
+                (old_key->lanes[lane].dst_md_index ==
+                 new_key->lanes[reuse_lane].dst_md_index)) {
```

Comment:
shall we check it as follows:
```
(old_key->lanes[lane].dst_md_index == UCP_NULL_RESOURCE) ||
(new_key->lanes[reuse_lane].dst_md_index == UCP_NULL_RESOURCE) ||
(old_key->lanes[lane].dst_md_index == new_key->lanes[reuse_lane].dst_md_index)
```
if not, pls add the following assertions:
```
ucs_assert(old_key->lanes[lane].dst_md_index != UCP_NULL_RESOURCE);
ucs_assert(new_key->lanes[reuse_lane].dst_md_index != UCP_NULL_RESOURCE);
```

Reply:
@changchengx wdyt?

Reply:
@dmitrygx Done